cpp/src/indexing/construction/detail/phase_1.cuh

```diff
@@ -287,6 +287,22 @@ inline auto make_full_levels(cudf::column_view const &x,
                                               thrust::plus<uint32_t>(),
                                               stream);
 
+  // Optimization: return early if all the nodes are top-level leaf children of the root.
+  //
+  // This allows us to avoid building the "full" quads for each level only to turn around and
+  // remove them for having no children. This scenario causes the `thrust::transform` call in
+  // `construct_non_leaf_indicator()` to launch on zero elements.
+  // if (num_top_quads >= static_cast<cudf::size_type>(quad_keys.size())) {
```

Comment:
Remove commented out code...

Reply:
Ah yes, I pushed this yesterday so @zhangjianting could try out some different fixes. Forgot to change the title back to WIP.